vm/src/builtins/property.rs

```diff
@@ -154,6 +155,23 @@ impl PyProperty {
         *self.doc.write() = value;
     }
 
+    #[pymethod(magic)]
+    fn set_name(&self, args: PosArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let arg_len = args.into_vec().len();
+
+        if arg_len != 2 {
+            Err(vm.new_exception_msg(
+                vm.ctx.exceptions.type_error.to_owned(),
+                format!(
+                    "__set_name__() takes 2 positional arguments but {} were given",
+                    arg_len
+                ),
```

Comment:
This message is common python type error which we are not handling well. We'd better to find enhance the general argument error handling instead of this way.

Reply:
I am sorry, the given type is `PosArgs`. So no way to do it in shared code.